### PR TITLE
MDP-188: Header component

### PR DESCRIPTION
**Pull request by PavolKiss**

### CODE REVIEW

File: src/Router.tsx [hunk 1]
```diff
@@ -12,6 +13,7 @@ export default () => {
       <Example path="foo" />
       <Counter path="counter" />
       <Testing path="testing" />
+      <Header path="header" />
```
Comment (pggl) [inline src/Router.tsx]: why is Header component here?

File: package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
     "awesome-typescript-loader": "^5.2.1",
     "jest": "^24.5.0",
     "parcel-bundler": "^1.9.7",
+    "postcss-import": "^12.0.1",
```
Comment (pggl) [inline package.json]: have you merge the master branch?

File: src/components/Header/header.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React, { FC } from "react";
+import chatImg from "../../static/svg/chatLogo.png";
+import Menu from "./Menu";
+import UserButton from "./UserButton";
+import ImageHeader from "./ImageHeader";
+import Navigation from "./Navigation";
+import { RouteComponentProps } from "@reach/router";
+
+const Header: FC<RouteComponentProps> = () => (
```
Comment (pggl) [inline src/components/Header/header.tsx]: remove RouteComponentProps and FC

File: src/components/Header/header.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      <Menu />
+      <ImageHeader src={chatImg} width={73} className="h-6 lg:ml-12" />
+      <Navigation />
+      <UserButton />
+    </div>
+    <div className="bg-lilac h-px" />
+  </div>
+);
+
+export default Header;
```
Comment (pggl) [inline src/components/Header/header.tsx]: filename should have Capital letter

File: src/components/Header/menu.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+import menuIcon from "../../static/svg/menu.svg";
+import ImageHeader from "./ImageHeader";
+
+const Menu = () => (
+  <div className="inline-block lg:hidden">
+    <ImageHeader src={menuIcon} className="ml-4" />
+  </div>
+);
+
+export default Menu;
```
Comment (pggl) [inline src/components/Header/menu.tsx]: filename should have capital letter

File: src/components/Header/navigation.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    <Link to="#" className="link pr-8">
+      Rozvrh
+    </Link>
+    <Link to="#" className="link">
+      Clanky
+    </Link>
+  </div>
+);
+
+export default Navigation;
```
Comment (pggl) [inline src/components/Header/navigation.tsx]: filename should have capital letter

File: src/components/Header/ImageHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import React, { FC } from "react";
+
+interface IProps {
+  src: string;
+  width?: number;
```
Comment (pggl) [inline src/components/Header/ImageHeader.tsx:5]: if this is optional we need some default value

File: src/components/Header/UserButton.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import userIcon from "../../static/svg/userIcon.svg";
+import ImageHeader from "./ImageHeader";
+import UserProfile from "./UserProfile";
+import UserMenu from "./UserMenu";
+import { Link } from "@reach/router";
+
+const UserButton = () => (
```
Comment (pggl) [inline src/components/Header/UserButton.tsx]: I think this component have bad name, there are links here

File: src/components/Header/UserButton.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  <div className="flex flex-row w-auto lg:mr-10 w-48 flex flex-row">
+    <div className="w-8 h-8 rounded-lg mr-4 flex items-center justify-center bg-lilac">
+      <UserMenu />
+      <ImageHeader src={userIcon} className="flex flex-row h-4 w-4 lg:hidden" />
+    </div>
+    <UserProfile />
+    <div className="flex flex-col overflow-auto z-10 absolute">
```
Comment (pggl) [inline src/components/Header/UserButton.tsx]: extract this into separate component

File: src/components/Header/UserMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+const UserMenu = () => (
+  <div className="hidden lg:inline-block">
+    <div className="font-gilbert font-bold text-lg">FS</div>
```
Comment (pggl) [inline src/components/Header/UserMenu.tsx]: we need props for this "FS"

File: src/components/Header/UserProfile.tsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+
+const UserProfile = () => (
+  <div className="hidden lg:inline-block">
+    <div className="h-5 font-gilbert text-base font-bold">
+      Feri Schwartzeneger
```
Comment (pggl) [inline src/components/Header/UserProfile.tsx]: we need props for this name

File: src/components/Header/UserProfile.tsx [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+    </div>
+    <div className="text-blue font-lato text-xs font-bold">Kredit 10 €</div>
```
Comment (pggl) [inline src/components/Header/UserProfile.tsx]: also here for credit

File: src/components/Header/imageHeader.tsx [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+  src: string;
+  width?: number;
+  className?: string;
+}
+
+const ImageHeader: FC<IProps> = ({ src, width, className }) => (
+  <img src={src} width={width} className={className} />
+);
+
+export default ImageHeader;
```
Comment (pggl) [inline src/components/Header/imageHeader.tsx]: old imageHeader is still there delete it

File: src/components/Testing.tsx [hunk 14]
```diff
@@ -1,24 +1,7 @@
 import React, { FC } from "react";
-import styled from "@emotion/styled";
 import { RouteComponentProps } from "@reach/router";
-import ButtonExample from "./ButtonExample";
+import Header from "./Header/Header";
```
Comment (pggl) [inline src/components/Testing.tsx]: remove this

File: src/components/Testing.tsx [hunk 15]
```diff
@@ -1,24 +1,7 @@
...
-      <ButtonExample text="Nákup kreditov" backgroundColor="black" />
-      <ButtonExample text="Nákup kreditov" backgroundColor="orangeRed" />
-      <ButtonExample text="Nákup kreditov" backgroundColor="shamrockGreen" />
-      <div className="p-4 bg-red border-red-dark border border-solid font-gilbert">
-        test tailwind
-      </div>
-    </Wrapper>
-  </div>
-);
+const Testing: FC<RouteComponentProps> = () => <Header />;
```
Comment (pggl) [inline src/components/Testing.tsx]: remove this and add `<div>testing</div>`

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/style/custom.css [hunk 16]
```diff
@@ -1,3 +1,7 @@
 .link {
   @apply font-gilbert no-underline inline-block font-bold text-base text-black;
 }
+
+.userProfileLinks {
+  @apply font-lato no-underline text-black
+}
```
Comment (garySiladi) [inline src/style/custom.css]: can you add a newline at the end of the file

**Review by garySiladi — DISMISSED** [review]
fine by me, contact Peter for a re-review

File: src/index.html [hunk 17]
```diff
@@ -2,7 +2,7 @@
   <head>
     <title>CHAT</title>
     <meta charset="UTF-8" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (pggl) [inline src/index.html]: revert this change

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/index.html (hunk 17 above)
Comment (PavolKiss) [reply src/index.html]: reverted.

**Review by pggl — APPROVED** [review]
Good work

**Pull request merged**